nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -985,3 +986,34 @@ def _fetch_next_batch(self, iterator: Iterator) -> None:
             assert isinstance(dataloader, Sized)  # `_has_len` is True
             self.done = self.fetched >= len(dataloader)
         self.on_fetch_end(batch, start_output)
+
+
+class CustomProgressBar(TQDMProgressBar):
+    """
+    Add a CustomProgressBar to remove 's/it' and add train_step_timing for megatron models
+    """
+
+    def __init__(self, cfg):
+        super().__init__()
+        self.cfg = cfg
+
+    def init_train_tqdm(self):
+        """
+        Override progress bar total to reflect total microbatches as PTL 2.0 progress bar indicates total microbatches, and bar format 
+        to remove 's/it' from the progress bar for megatron models
+        """
+        self.bar = super().init_train_tqdm()
+        # TODO athitten: Remove the below line once PTL fixes the logic for progress bar with dataloader_iter
```

Comment:
what if someone sets epochs instead of setting max_steps, would this still work?

Reply:
The CustomProgressBar is being restricted only for Megatron and I was under the impression we always use max_steps and don't have concept of epochs. Let me know if I am missing something here.

Reply:
We don't need to update bar total with latest fix, so this shouldn't be a concern.